Source/Signal.swift

```diff
@@ -109,3 +109,73 @@ extension SignalType where T: SequenceType {
         }
     }
 }
+
+// MARK: Boolean Operations
+
+extension SignalType where T: BooleanType {    
+    /// Returns a signal that inverts each value sent on self
+    @warn_unused_result(message="Did you forget to call `observe` on the signal?")
+    public func not() -> Signal<Bool, E> {
```

Comment:
Actually I'd like to return `Signal<T, E>`, but the `!` operator has a signature of `!<T : BooleanType>(_ a: T) -> Bool` already...